cli/src/main/kotlin/tools/samt/cli/CliFileResolution.kt

```diff
@@ -0,0 +1,35 @@
+package tools.samt.cli
+
+import tools.samt.common.DiagnosticController
+import tools.samt.common.SourceFile
+import java.io.File
+
+internal fun List<String>.readSamtSourceFiles(controller: DiagnosticController): List<SourceFile> {
+    val files = map { File(it) }.ifEmpty { gatherSamtFiles(controller.workingDirectoryAbsolutePath) }
```

Comment:
could use java.nio.Path instead of java.io.File

https://www.baeldung.com/java-path-vs-file

Reply:
I can't easily walk down the `Path` without converting it to a `File` first, and converting both ways seems strange. Also most code just changes in such a way, which is why I would leave the `File` for now